File: Java-Binary-Calculator/src/Model/CalculatorLogic.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Model;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

import javax.swing.JOptionPane;

/**
 *
 * @author test
 */
public class CalculatorLogic {
    public String binaryLogic(String num) throws NumberFormatException
    {
        //String num = txt1.getText();
        int numx = Integer.parseInt(num);
        HashMap<Integer, Integer> data = new HashMap<Integer, Integer>();
            
        //find boolian numbers
        for(int c=0; numx!=0;c++)
        {
            int numy = numx%2;
            numx = numx/2;
            data.put(c+1, numy);
        }
            
        //Make arraylist to hold boolian numbers
        List<String> output = new ArrayList<String>();
            
        //turn upside down the boolian number
        for(int d=data.size(); 0<d; d--)
        {   
            output.add(Integer.toString(data.get(d)));
        }
        
        //arraylist convert to String variable
        String final_output = String.join("",output);
        
        return final_output;
    }
    
    public String realNumLogic(String num) throws NumberFormatException
    {  
        //multiplicator( 1, 2, 4 ,8...)
        int mul = 1;
        int total = 0;
        
        //turn upside down boolean number
        for(int c=num.length(); 0<c; c--)
        {
            // calculate real number
            int numtemp = Character.getNumericValue(num.charAt(c-1));
            if (mul==1){
                total=total+mul*numtemp;
                 mul++;
            }
            else{
                total=total+mul*numtemp;
                mul=mul*2;
            }
            
        }
        return (Integer.toString(total));
    }
}

File: Kotlin-Binary-Calculator/app/src/main/java/com/example/kotlin_binary_calculator/CalculatorLogic.kt
package com.example.kotlin_binary_calculator

class CalculatorLogic {

    fun decimalToBinary (num : String) : String {
        var number : Int = num.toInt()
        var numx : Int = 0
        var data = ArrayList<Int>()
        var rnum : String = ""

        while (number != 0) {
            numx = number%2
            number = number/2
            data.add(numx)
        }

        for (i in (0..data.size-1).reversed()) {
            rnum += data.get(i).toString()
        }

        return rnum
    }

    fun decimalToOctal (num : String) : String {
        var number : Int = num.toInt()
        var numx : Int = 0
        var data = ArrayList<Int>()
        var rnum : String = ""

        while (number != 0) {
            numx = number%8
            number = number/8
            data.add(numx)
        }

        for (i in (0..data.size-1).reversed()) {
            rnum += data.get(i).toString()
        }

        return rnum
    }

    fun binaryToDecimal (num : String): String {
        //multiplicator( 1, 2, 4 ,8...)
        var pos : Int = 1
        var total : Int = 0

        //turn upside down boolean number
        for(c in (1..num.length).reversed())
        {
            // calculate real number
            var numtemp = Character.getNumericValue(num.get(c-1));
            if (pos==1){
                total = total + pos * numtemp;
                pos++;
            }
            else{
                total = total + pos * numtemp;
                pos *= 2;
            }

        }
        return (total.toString());
    }

    // 0 <= number <= 7
    fun octalToDecimal (num : String) : String {
        var pos : Int = 1   // ( 1, 8, 64...)
        var total : Int = 0

        // turn upside down octal number
        for(c in (1..num.length).reversed())
        {
            // calculate real number
            var numtemp = Character.getNumericValue(num.get(c-1));
            total = total + pos * numtemp;
            pos *= 8;

        }
        return (total.toString());
    }

}